code/ui/blocks/src/components/TableOfContents.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect } from 'react';
+import type { FC, ReactElement } from 'react';
+import { styled } from '@storybook/theming';
+import tocbot from 'tocbot';
+
+export interface TocParameters {
+  /** CSS selector for the container to search for headings. */
+  contentsSelector?: string;
+
+  /**
+   * When true, hide the TOC. We still show the empty container
+   * (as opposed to showing nothing at all) because it affects the
+   * page layout and we want to preserve the layout across pages.
```

Comment:
Do I understand this correctly, that if a user has `toc` in project annotations, and for a single component have `disable`, we always show the TOC container (but sometimes empty), so that layout won't shift?

But if the user doesn't have `toc` set global, but just for a single story, then we'll have layout shifts?

I think this is a good solution, I just want to make sure my understanding is correct.

Reply:
I believe so. This is @cdedreuille 's comment I believe.